File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;


public class Main {
    public void solution() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;
        
        Stack<Integer> stk = new Stack<>();
        int N = Integer.parseInt(br.readLine());
        for (int i=0;i<N;i++){
            st = new StringTokenizer(br.readLine());
            String q = st.nextToken();
            
            int state;
            switch (q){
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    stk.push(num);
                    break;
                case "pop":
                    state = stk.empty() ? -1 : stk.pop();
                    bw.write(state + "\n");
                    break;
                case "size":
                    state = stk.size();
                    bw.write(state + "\n");
                    break;
                case "empty":
                    state = stk.empty() ? 1 : 0;
                    bw.write(state + "\n");
                    break;
                case "top":
                    state = stk.empty() ? -1 : stk.peek();
                    bw.write(state + "\n");
                    break;
            }
        }
        
        br.close();
        bw.flush();
        bw.close();
    }
    
    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}

File: 백준/Silver/10828. 스택/스택.py
import sys

input = sys.stdin.readline

N = int(input())

stack = []
for _ in range(N):
    query = input().strip().split()

    if query[0] == "push":
        stack.append(query[1])

    elif query[0] == "pop":
        if not stack:
            print(-1)

        else:
            print(stack.pop())

    elif query[0] == "size":
        print(len(stack))

    elif query[0] == "empty":
        print(1 if not stack else 0)

    else:
        if not stack:
            print(-1)

        else:
            print(stack[-1])